### PR TITLE
feat: upgrade to response-rejection plugin

### PR DESCRIPTION
**Pull request by carlbrugger**
**Guide update:** https://github.com/FlatFilers/Guides/pull/955
**Kitchen Sink update:** https://github.com/FlatFilers/flatfile-docs-kitchen-sink/pull/34

This PR adds better handling of submission status to the `@flatfile/util-response-rejection` plugin by providing webhook with 2 options:
    - Option 1: Delete all successful records and leave only rejections
    - Option 2 _(default)_: Create new (readonly) submission status column that marks each record as "submitted" or "rejected"

Also, the webhook is given the option to set a response message.

[Preview video](https://www.loom.com/share/c97985c0ad1849169ff2724ffae1775e?sid=2e2d23c8-6909-44ad-a7ec-7ce0b2cada12)

Future updates:
- Next URL will open in a new tab. Will be fixed soon.
- Forced page refresh when column is added. Will be fixed soon.

## Example
Clone https://github.com/carlbrugger/bun-webhook

Run:
```
bun
bun start
```
This will start a webhook server on `localhost:1234`.

Paste in `listener.ts`:
```
import type { FlatfileListener } from '@flatfile/listener'
import { configureSpace } from '@flatfile/plugin-space-configure'
import { webhookEgress } from '@flatfile/plugin-webhook-egress'

export default function (listener: FlatfileListener) {
  listener.use(
    webhookEgress(
      'workbook:submitActionFg',
      'http://localhost:1234/reject-non-flatfile-emails'
    )
  )

  listener.use(
    configureSpace({
      workbooks: [
        {
          name: 'Playground',
          sheets: [
            {
              name: 'Contacts',
              slug: 'contacts',
              allowAdditionalFields: true,
              fields: [
                {
                  key: 'firstName',
                  type: 'string',
                  label: 'First Name',
                },
                {
                  key: 'lastName',
                  type: 'string',
                  label: 'Last Name',
                },
                {
                  key: 'email',
                  type: 'string',
                  label: 'Email',
                },
              ],
            },
          ],
          actions: [
            {
              operation: 'submitActionFg',
              mode: 'foreground',
              label: 'Submit data',
              type: 'string',
              description: 'Submit this data to a webhook.',
              primary: true,
            },
          ],
        },
      ],
    })
  )
}
```

### CODE REVIEW

**Review by ashleygmulligan2 — DISMISSED** [review]
concern of this being a breaking change in the utl. Types names have changed. Also, the rejection handler now returns a job response object rather than a total number of rejections as before. This should be a major release. Since the plugin itself isn't impacted in that way, patch seems fine. Tho this is more than a patch/fix, and introduces new functionality, so maybe minor is better here.

File: utils/response-rejection/src/index.ts [hunk 1]
```diff
@@ -1,63 +1,156 @@
...
+}
 
-  return totalRejectedRecords
+function getNext(totalRejectedRecords): Flatfile.JobOutcomeNext | undefined {
+  return totalRejectedRecords > 0
+    ? {
+        type: 'url',
+        url: '?searchField=submissionStatus&searchValue=rejected',
+        label: 'See rejections',
+      }
```
Comment (carlbrugger) [inline utils/response-rejection/src/index.ts:57]: Change this to use : `{ next: { type: 'id', id: 'us_sp_12345', query: 'filter=error'}}`

File: plugins/webhook-egress/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   },
   "license": "ISC",
   "dependencies": {
-    "@flatfile/api": "^1.5.37",
+    "@flatfile/api": "^1.5.45",
```
Comment (bangarang) [inline plugins/webhook-egress/package.json:30]: Should we bump this to `^1.6.0`

File: utils/response-rejection/src/index.ts [hunk 3]
```diff
@@ -1,63 +1,162 @@
...
+  return sheetRejections.rejectedRecords.length
+}
+
+async function addSubmissionStatusField(sheetId: string): Promise<void> {
+  try {
+    const { data: sheet } = await api.sheets.get(sheetId)
+    if (
+      !sheet.config.fields.some((field) => field.key === 'submissionStatus')
+    ) {
+      await api.sheets.addField(sheet.id, {
```
Comment (bangarang) [inline utils/response-rejection/src/index.ts:88]: Should we remove this field from the sheet when submitting? I'm sure they could strip it out if they wanted but could be nice if it was cleaned up as an option

**Review by bangarang — APPROVED** [review]
Awesome work!

File: plugins/webhook-egress/package.json (hunk 2 above)
Comment (carlbrugger) [reply plugins/webhook-egress/package.json]: Yes

**Pull request merged**